content/post/2020-01-15-examples/index.Rmd

```diff
@@ -0,0 +1,148 @@
+```
+
+or 
+
+```r
+library("pkgsearch)
+example("pkg_search")
+```
+
+As a side-effect, this makes all objects created in the example(s) available in your workspace.
+
+### How to run specific examples from a man page
+
+If you want to run only one of the examples you're reading
+
+* You can select the lines and then run Command+Enter (Ctrl+Enter on Linux), be it in the R GUI or in RStudio. Below is a screenshot created by [Mine Çetinkaya-Rundel](https://twitter.com/minebocek) when she curated the [We are R-Ladies Twitter account](https://twitter.com/WeAreRLadies).
+
+```{r echo=FALSE}
+
+blogdown::shortcode("tweet", "1064734591641104384")
+```
+
+* From an online `pkgdown` website, you'd use the copy button and then paste the result in your R console.
+
+## How to write examples?
+
+In an Rd file, examples are inside `\examples{}` e.g.
+
+```r
+\examples{
+  # basic usage of foo
+  foo(basic = TRUE)
+  # elaborate usage of foo
+  foo(basic = FALSE)
+}
+```
+
+But usually nowadays docs are generated by `roxygen2` so your examples can either
+
+* exist after the `@examples` tag
+
+```r
+#' @examples
+#' # basic usage of foo
+#' foo(basic = TRUE)
+#' # elaborate usage of foo
+#' foo(basic = FALSE)
+```
+
+* exist in an R script that you'd source from the docs i.e.
+
+```r
+#' @example man/examples/foo.R
+```
+
+and in `man/examples/foo.R`
+
+```r
+# basic usage of foo
+foo(basic = TRUE)
+# elaborate usage of foo
+foo(basic = FALSE)
+``` 
+
+The former solution (`@examples` tag: examples with other docs for the function inside the function R file) means you can work on the function code and examples at the same time more smoothly, but all your lines start with `#'`; the latter solution (`@example` tag without an s: examples in dedicated files) means you can write examples as usual R code without `#'`. 
+
+## How to know there are enough examples?
+
+You could judge whether the man page of each function shows the most important behaviours by reading said man pages. You could also use the output of [`covr::package_coverage(<path-to-your-package>, type = "examples")`](https://covr.r-lib.org/reference/package_coverage.html) to reveal major gaps!
+
+## How to safeguard examples?
+
+Examples are supposed to be _executable_ i.e. to run without error. As writing in ["Writing R Extensions"](https://cran.r-project.org/doc/manuals/R-exts.html#index-_005cdontrun), _"By default, text inside `\examples{}` will be displayed in the output of the help page and run by `example()` and by `R CMD check`."_. `R CMD check`'s running example could be described as a [_smoke test_](https://en.wikipedia.org/wiki/Smoke_testing_(software)).
+
+Now comes one of the trickiest aspects of creating examples for the manual pages: how to execute them only when they work? We're not encouraging cheating but
+
+* Examples can show an error, to exemplify the behaviour of the function in such cases.
+* Examples can need user setup such as storing an API key in `.Renviron`.
```

Comment:
can need -> may need
storing an API key in `.Renviron` -> having an API key available